Radio.m

```diff
@@ -1445,14 +1591,66 @@ - (void) parseGpsToken: (NSScanner *) scan {
     
 }
 
-
+//
+// Parse Profile tokens
+//     called on the GCD thread associated with the GCD tcpSocketQueue
+//
+//     format: <apiHandle>|profile <profileType> list=<value>^<value>^...<value>^
+//                           OR
+//     format: <apiHandle>|profile <profileType> current=<value>
+//
+//     scan is initially at scanLocation = 17, start of the <profileType>
+//     "<apiHandle>|profile " has already been processed
+//
 - (void) parseProfileToken: (NSScanner *) scan {
-    
+    // get the Profile type
+    NSString *profileType;
+    [scan scanUpToString:@" " intoString: &profileType];
+    // skip the space
+    [scan scanString:@" " intoString: nil];
+    
+    // get the Sub type
+    NSString *profileSubType;
+    [scan scanUpToString:@"=" intoString: &profileSubType];
+    // skip the "="
+    [scan scanString:@"=" intoString: nil];
+    
+    // get the remainder of the command
+    NSString *remainderOfCommand;
+    [scan scanUpToString:@"\n" intoString: &remainderOfCommand];
+    
+    // List or Current value?
+    if ([profileSubType isEqualToString: @"list"]) {
+        // it's the List, separate the components
+        NSMutableArray *profileNames;
+        profileNames = [[remainderOfCommand componentsSeparatedByString:@"^"] mutableCopy];
+        // remove the last (empty) string
+        [profileNames removeLastObject];
+        // save it in the appropriate property
+        if ([profileType isEqualToString: @"global"]) {
+            @synchronized (self.globalProfiles) {
+                _globalProfiles = profileNames;
```

Comment:
this needs to be an updateAndNotify so that a client of the model can KVO for changes on the array.

Although Flex hasn't implemented profiles this way, I think from a clients point of view there should be a ProfileCreated and a ProfileDeleted notification.  Its possible that someone may be KVO'ing the array and if so, they need to be notified before the array is changed so that the observer can de-register or else an exception will be thrown.  

I'm surprised that they didn't add a status notification telling a client that a profile has been removed rather than just dumping out a list of all the names.

Also, I notice that flex uses the caret ^ to replace spaces in profile names.  That should be handled transparently to our client here.  Both for setting the names in the arrays plus for removing spaces when the profile name is sent to the radio.